hooks/useFetch.js

```diff
@@ -73,27 +76,20 @@ const getURL = (url) =>
     materialize(),
     filter((i) => !isComplete(i)),
   )
+}
 
 const cache = {}
 
 const fetchData = (url) => {
   if (url in cache) return cache[url]
 
-  const loading = new BehaviorSubject(false)
-
-  const data = new BehaviorSubject(url).pipe(
-    tap(() => loading.next({ url: url, status: true })),
-    switchMap(getURL),
-    tap(() => loading.next({ url: url, status: false })),
-    shareReplay(1),
-  )
-
-  const refresh = combineLatest(of(url), interval(1000)).pipe(
-    pluck(0),
-    tap(data),
+  const refresher = timer(0, 1000).pipe(
+    switchMap(() => getURL(url)),
+    publish(),
```

Comment:
Okay, so this sets the timer which fires immediately and every second. It doesn't actually do anything though until something subscribes because of the publish? Why is the publish required if observables don't go hot until the subscription normally?

Reply:
The publish is required to turn the timer into something multiple subscribers can subscribe to. Think about it like a multiplexer. The `refresher` stream which is subscribed to the timer stream won't start firing until it has at least 1 subscriber which has also called `connect` on the subscription to turn it hot. To avoid requiring subscribers having to call connect/disconnect is why the next line leverages the `refCount`er.